### PR TITLE
feat: Filter conversation by favorites - WPB-7324

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7324" title="WPB-7324" target="_blank">TaskWPB-7324</a>  [iOS] : Header : Filter by Favourites
  </td></table>
  <br />
 

This pull request introduces a new functionality to filter conversations in the conversation list using a favorites filter. Additionally, it addresses an issue where the filter button would move to the wrong position upon selection.

That's fixed with this line inside the setupRightNavigationBarButtons method:

```swift
// Trigger a layout update to ensure the correct positioning
// of the add conversation button and filter button
// when the filter button is tapped.
self.view.setNeedsLayout()
```

1. Favorites Filter Functionality:

- Added the ability to filter conversations based on their favorites status within the conversation list.
- Users can now easily view and manage their favorite conversations by applying this filter.

2. UI Fix for Filter Button:

- Fixed a bug that caused the filter button to shift to an incorrect position after tapping and making a selection.
- Ensured that the filter button remains in the correct position after any interaction.

--

Note: Snapshot Tests are going to be added at a later point when I finish with all of the UI changes.

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift [hunk 1]
```diff
@@ -21,11 +21,11 @@ import WireCommonComponents
 import WireDataModel
 import WireSyncEngine
 
-extension ConversationListViewController {
+enum FilterType {
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift]: If moved out from the scope of `ConversationListViewController`, could we name it more specifically, please?
Like `ConversationFilterType`?

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift [hunk 2]
```diff
@@ -219,15 +230,25 @@ extension ConversationListViewController {
...
 
+    /// Method to apply the selected filter and update the UI accordingly
+    private func applyFilter(_ filter: FilterType) {
+        self.listContentController.listViewModel.selectedFilter = filter
+        self.setupRightNavigationBarButtons()
+
+        // Trigger a layout update to ensure the correct positioning
+        // of the add conversation button and filter button
+        // when the filter button is tapped.
+        self.view.setNeedsLayout()
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift]: I suggest to move this line into `setupRightNavigationBarButtons` (as last line).

**Review by markusfassbender — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift]: Done: https://github.com/wireapp/wire-ios/commit/672ad00ac9de6c1314e27ffc2fadf292e4002ca0

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift]: Done: https://github.com/wireapp/wire-ios/commit/672ad00ac9de6c1314e27ffc2fadf292e4002ca0

Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+NavigationBar.swift]: Done: https://github.com/wireapp/wire-ios/commit/672ad00ac9de6c1314e27ffc2fadf292e4002ca0

**Review by caldrian — APPROVED** [review]

**Pull request merged**